boot/zephyr/CMakeLists.txt

```diff
@@ -70,68 +70,88 @@ set(TINYCRYPT_DIR "${MCUBOOT_DIR}/ext/tinycrypt/lib")
+	flash_map_legacy.c
+	)
 endif()
 
 # Generic bootutil sources and includes.
-target_include_directories(app PRIVATE "${BOOT_DIR}/bootutil/include")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/loader.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/bootutil_misc.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/image_validate.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/encrypted.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/image_rsa.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/image_ec256.c")
-target_sources(app PRIVATE "${BOOT_DIR}/bootutil/src/caps.c")
+zephyr_library_include_directories(${BOOT_DIR}/bootutil/include)
+zephyr_library_sources(
+  ${BOOT_DIR}/bootutil/src/loader.c
+  ${BOOT_DIR}/bootutil/src/bootutil_misc.c
+  ${BOOT_DIR}/bootutil/src/image_validate.c
+  ${BOOT_DIR}/bootutil/src/encrypted.c
+  ${BOOT_DIR}/bootutil/src/image_rsa.c
+  ${BOOT_DIR}/bootutil/src/image_ec256.c
+  ${BOOT_DIR}/bootutil/src/caps.c
+  )
 
 if(CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256)
   # When using ECDSA signatures, pull in our copy of the tinycrypt library.
-  target_include_directories(app PRIVATE "${BOOT_DIR}/zephyr/include")
-  target_include_directories(app PRIVATE "${TINYCRYPT_DIR}/include")
-  target_include_directories(app PRIVATE "${MBEDTLS_ASN1_DIR}/include")
-
-  target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/ecc.c")
-  target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/ecc_dsa.c")
-  target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/sha256.c")
-  target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/utils.c")
+  zephyr_library_include_directories(
+	 ${BOOT_DIR}/zephyr/include
+	   ${TINYCRYPT_DIR}/include
+	${MBEDTLS_ASN1_DIR}/include
+	)
+
+  zephyr_library_sources(
+	${TINYCRYPT_DIR}/source/ecc.c
+	${TINYCRYPT_DIR}/source/ecc_dsa.c
+	${TINYCRYPT_DIR}/source/sha256.c
+	${TINYCRYPT_DIR}/source/utils.c
+
+	# Additionally pull in just the ASN.1 parser from mbedTLS.
+	${MBEDTLS_ASN1_DIR}/src/asn1parse.c
+	${MBEDTLS_ASN1_DIR}/src/platform_util.c
+	)
 
   # Since here we are not using Zephyr's mbedTLS but rather our own, we need
   # to set MBEDTLS_CONFIG_FILE ourselves. When using Zephyr's copy, this
   # variable is set by its Kconfig in the Zephyr codebase.
-  target_compile_definitions(app PRIVATE MBEDTLS_CONFIG_FILE="${CMAKE_CURRENT_LIST_DIR}/include/mcuboot-mbedtls-cfg.h")
-  # Additionally pull in just the ASN.1 parser from mbedTLS.
-  target_sources(app PRIVATE "${MBEDTLS_ASN1_DIR}/src/asn1parse.c")
-  target_sources(app PRIVATE "${MBEDTLS_ASN1_DIR}/src/platform_util.c")
+  zephyr_library_compile_definitions(
+	MBEDTLS_CONFIG_FILE="${CMAKE_CURRENT_LIST_DIR}/include/mcuboot-mbedtls-cfg.h"
+	)
 elseif(CONFIG_BOOT_SIGNATURE_TYPE_RSA)
   # Use mbedTLS provided by Zephyr for RSA signatures. (Its config file
   # is set using Kconfig.)
   zephyr_include_directories(include)
-  target_include_directories(app PRIVATE $ENV{ZEPHYR_BASE}/ext/lib/crypto/mbedtls/include)
```

Comment:
why this line is not rewritten?

Reply:
Because it is redundant.

When MBEDTLS is enabled the app is automatically linked with the MBEDTLS library. The MBEDTLS library again has this include dir in it's INTERFACE properties, so the app get's this include path from there.

This is implemented here:

https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/app/boilerplate.cmake#L479